sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureResource.java

```diff
@@ -34,10 +34,10 @@
  * It also serves as the interface to Storage clients. Any operation that needs to use a client should first build an
  * AzureResource using a path and then use the getter to access the client.
  */
-class AzureResource {
+final class AzureResource {
     private final ClientLogger logger = new ClientLogger(AzureResource.class);
 
-    static final String DIR_METADATA_MARKER = "is_hdi_folder";
+    static final String DIR_METADATA_MARKER = "hdi_isfolder";
```

Comment:
Can we possibly swap all instances of NIOs DIR_METADATA_MARKER for the one in common? I just merged a PR today that added the constant for a datalake feature

Reply:
Definitely will do that.